### ISSUE TITLE
Interpolation in Variables Section

### ISSUE THREAD

**Issue by jenpet** [tj/robo#44]
Thanks for that nifty tool! It triggered me to get rid of `Makefiles` in smaller projects. I saw that there is a [ticket for includes](#5 ) which I also sort of "bumped" since I think it could be a real benefit when re-using common parts in several projects.

What do you think about the possibility to interpolate the values of the `variable` section? A construct like the following snippet would definitely help for more flexibility.

```
task:
   command: echo {{ .bar }} // Hello World e9dsasdh1

variables:
   hash: $(git rev-parse --short HEAD)
   foo: Hello
   bar: {{ .foo }} World {{ .hash }}
```

If I need those variables across multiple tasks it would be great if I could just reuse them from the "global scope" instead of exporting the via env vars first with a separate task. Am I missing something in the current feature set or would this be also beneficial for others?

**Comment by yields** [comment]
That’s an interesting use case, I think that’s a great idea. We would probably have to add it just before interpolating the [tasks](https://github.com/tj/robo/blob/master/config/config.go#L31).

Wanna take a stab at it?

**Comment by jenpet** [comment]
Cool, I think it would act pretty similar to the "global vars section" of the Makefile approach. 

Yeah, will try to dig into this a bit. If I get stuck at some point I'd reach out to you guys.

**Comment by yields** [comment]
Thanks @jenpet! -- released [v0.6.0](https://github.com/tj/robo/releases/tag/v0.6.0).
